Update readme

josh146 opened the pull request:
Updates the QML readme to be more visually appealing, and to include more links back to the QML website. This should make the demonstration webpage more discoverable for users who find the repository via GitHub, as well as better articulate what this repo is.

In addition:

* Adds a 'contributions welcome' badge
* All screenshots are hyperlinks to their website location

The rendered readme can be viewed here: https://github.com/PennyLaneAI/qml/tree/update-readme

Screenshot:

<img src="https://user-images.githubusercontent.com/2959003/92303937-c537a280-efb8-11ea-8b9f-b3ce479d3aeb.png" width="700px">

thisac commented on README.md:
```diff
@@ -18,6 +36,13 @@ as well as Python code demos. Both can be accessed at
+
 ## Contributing
 
 You can contribute by submitting a demo implementing a recent
```
There's a line-break on line 60 that shouldn't be there. After `If you would like to write a demo...`.

josh146 replied:
good catch!

thisac commented on README.md:
```diff
@@ -105,7 +130,7 @@ there are a couple of guidelines to keep in mind.
   be set for reproducibility.
```
On line 127 above, punctuation is missing after `Any included text should be minimal and legible`.

thisac reviewed (approved):
Thanks @josh146. This looks great! Just found two very minor things (that seemed to be there before).

josh146 commented:
Thanks @thisac and @co9olguy. I've updated the PR with the suggested changes, and also added in two new bullet points:

* The difference between the `master` and `dev` branches as per @ixfoduap's suggestion

* How to build only a single tutorial/demo

thisac commented on README.md:
```diff
@@ -45,7 +45,7 @@ and can be downloaded as Jupyter notebooks and Python scripts.
 ## Contributing
 
-You can contribute by submitting a demo implementing a recent
+You can contribute by submitting a demo via pull request implementing a recent
```
:100:

thisac commented on README.md:
```diff
@@ -102,6 +101,10 @@ quantum machine learning paper/result.
+If your demo uses the latest release of PennyLane, simply make your PR against the
+`master` branch. If you instead require the cutting-edge development versions of
+PennyLane or any relevant plugins, make your PR against the `dev` branch instead.
+
```
Great addition!

ixfoduap replied:
Awesome! Thanks for including this!
